Mathlib/Algebra/Lie/Abelian.lean

```diff
@@ -138,6 +138,20 @@ protected theorem mem_ker (x : L) : x ∈ LieModule.ker R L M ↔ ∀ m : M, ⁅
   simp only [LieModule.ker, LieHom.mem_ker, LinearMap.ext_iff, LinearMap.zero_apply,
     toEnd_apply_apply]
 
+lemma _root_.LieIdeal.isLieAbelian_iff {I : LieIdeal R L} :
+    IsLieAbelian I ↔ I ≤ (toEnd R L I).ker := by
+  refine ⟨fun hI x hx ↦ LieHom.mem_ker.mpr ?_, fun h ↦ ⟨fun ⟨x, hx⟩ ⟨y, hy⟩ ↦ ?_⟩⟩
+  · ext y
+    have := IsTrivial.trivial (⟨x, hx⟩ : I) y
+    rw [LieIdeal.coe_bracket_of_module] at this
+    simp [this]
+  · simpa using LinearMap.congr_fun (h hx) ⟨y, hy⟩
+
+lemma bracket_eq_zero_of_mem {I : LieIdeal R L} (hI : IsLieAbelian I) {x : L}
+    (hx : x ∈ I) (y : I) :
+    ⁅x, y⁆ = 0 :=
+  (LieModule.mem_ker R L I x).mp ((LieIdeal.isLieAbelian_iff R L).mp hI hx) y
```

Comment:
I'm a bit dubious about this lemma since:

1. I think we should be consistent about how we supply `x` and `y`: either both of type `L` satisfying membership hypothesis in `I` or both of type `I` (currently the conventions are mixed).
2. This is just a restatement of `IsTrivial.trivial (⟨x, hx⟩ : I) (y : I)`

Do we definitely need this?